src/Tribe/Repository/Query_Filters.php

```diff
@@ -0,0 +1,353 @@
+<?php
+
+/**
+ * Class Tribe__Repository__Query_Filters
+ *
+ * @since TBD
+ */
+class Tribe__Repository__Query_Filters {
+
+	/**
+	 * @var array
+	 */
+	protected $query_vars = array(
+		'like'   => array(
+			'post_title'   => array(),
+			'post_content' => array(),
+			'post_excerpt' => array(),
+		),
+		'status' => array(),
+	);
+
+	/**
+	 * @var WP_Query
+	 */
+	protected $current_query;
+
+	/**
+	 * Filters the WHERE clause of the query to match posts with a field like.
+	 *
+	 * @since TBD
+	 *
+	 * @param string   $where
+	 * @param WP_Query $query
+	 *
+	 * @return string
+	 */
+	public function filter_by_like( $where, WP_Query $query ) {
+		if ( $query !== $this->current_query ) {
+			return $where;
+		}
+
+		if ( empty( $this->query_vars['like'] ) ) {
+			return $where;
+		}
+
+		foreach ( $this->query_vars['like'] as $field => $entries ) {
+			foreach ( $entries as $entry ) {
+				$where .= $this->and_field_like( $field, $entry );
+			}
+		}
+
+		return $where;
+	}
+
+	/**
+	 * Builds the escaped WHERE entry to match a field like the entry.
+	 *
+	 * @since TBD
+	 *
+	 * @param string $field
+	 * @param string $entry
+	 *
+	 * @return string|void
+	 */
+	protected function and_field_like( $field, $entry ) {
+		/** @var wpdb $wpdb */
+		global $wpdb;
+		$like = '%' . $wpdb->esc_like( $entry ) . '%';
```

Comment:
This would be a bit more flexible if we only included `%` if there wasn't an unescaped `%` already present (`\%`).

```php
$like = $wpdb->esc_like( $entry );

// If there isn't an unescaped %, let's auto-add some %
if ( ! preg_match( '/(?<!\\)%/', $like ) ) {
  $like = "%{$like}%";
}
```